File: Queues/QueueLinkedListDequeue.java
class Node
{
    int data;
    Node next;

    Node(int data)
    {
        this.data = data;
        this.next = null;
    }
}

class QueueLinkedListDequeue
{
    Node front;

    void dequeue()
    {
        if (front == null)
        {
            System.out.println("Queue Underflow");
            return;
        }

        System.out.println("Deleted " + front.data);
        front = front.next;
    }

    public static void main(String[] args)
    {
        QueueLinkedListDequeue q = new QueueLinkedListDequeue();
        q.front = new Node(10);
        q.front.next = new Node(20);
        q.front.next.next = new Node(30);

        q.dequeue();
        q.dequeue();
    }
}

File: Queues/queue_linkedlist_dequeue.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

front = Node(10)
front.next = Node(20)
front.next.next = Node(30)

def dequeue():
    global front
    if front is None:
        print("Queue Underflow")
        return
    print("Deleted", front.data)
    front = front.next

dequeue()
dequeue()
